src/main/java/reactor/netty/channel/MonoSendMany.java

```diff
@@ -176,14 +180,13 @@ public void onNext(I t) {
 				return;
 			}
 
-			if (terminalSignal != null) {
+			if (terminalSignal != null || this.s == Operators.cancelledSubscription()) {
```

Comment:
Why ? shouldn't cancellation be checked in the drainLoop for safety anyway ?

Reply:
The answer is - if we will not do that: 

The element goes into a queue - and then will be discarded later in else branch of trySchedule since WIP will be 1. Why 1 - because by the moment, drainLoop can exit, discarded what was seen in the queue, and simply return leaving WIP == 1. (yeah, this is basically the issue which leaves some numbers of elements undiscarded). Thus, the question is - do we wanna discard element earlier, or do we wanna have enqueued with volatile write and then 2 volatile reads in order to say - we have to discard?